src/Sulu/Bundle/PreviewBundle/Resources/js/containers/Preview/Preview.js

```diff
@@ -186,13 +189,13 @@ class Preview extends React.Component<Props> {
         this.localeDisposer = reaction(
             () => toJS(formStore.locale),
             () => {
-                this.previewStore.restart();
+                this.previewStore.restart(toJS(formStore.locale));
```

Comment:
move the toJS into the previewStore class. We also unpack the locale in the constructor there